### PR TITLE
Feature/xkcd source

### PR DESCRIPTION
**Pull request by carlank**

### CODE REVIEW

File: src/Source/Source.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        this.tags = tags;
+    }
+
+    /**
+     * Check if the source has any of the given tags
+     * @param {string[]} tags The tags we're looking for
+     * @return {boolean}      Whether one of the tags corresponds to our own
+     */
+    isRelevantToAnyOfThese(tags) {
+        for (let tag in this.tags) {
```
Comment (Stratadox) [inline src/Source/Source.js:15]: I'd put `throw 'Method not defined'` here instead tbh

Comment (Stratadox) [reply src/Source/Source.js:15]: Although, I guess it'd probably be the same for all of them

**Review by Stratadox — DISMISSED** [review]
Nice!

File: src/Bot.js [hunk 2]
```diff
@@ -91,6 +89,7 @@ class Bot {
      */
     chooseSourceFor(channel) {
         for (const source of this.sources) {
+            console.log(source)
```
[the remark this answers is not in the document]
Comment (carlank) [reply src/Bot.js:92]: Yeah, point

File: __tests__/fetching_conversation_starters.test.js [hunk 3]
```diff
@@ -15,9 +17,5 @@ test('fetches the source with the matching tag', () => {
         },
         tags: ['tag']
     });
-
-    bot.notify('test-channel', new Date(0));
-    bot.reviveChannels(new Date(15 * 1000));
-
-    expect(message).toBe('expected');
+    expect(bot.chooseSourceFor(bot.channels.get('test-channel'))).toBe(correctSource);
```
Comment (Stratadox) [inline __tests__/fetching_conversation_starters.test.js:20]: Oh hmm that's kinda testing the internals

Comment (carlank) [reply __tests__/fetching_conversation_starters.test.js:20]: It is. If you have thoughts on a better way to do it, I'd love it

**Pull request merged**